### PR TITLE
clang/AMDGPU: Emit grid size builtins with range metadata

### PR DESCRIPTION
**Pull request by arsenm**
These cannot be 0.

### CODE REVIEW

**Comment by arsenm** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/113038?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#113038** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/113038?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#113019** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/113019?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#113018** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/113018?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @arsenm and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by jhuber6 — APPROVED** [review]

File: clang/test/CodeGenOpenCL/builtins-amdgcn.cl [hunk 1]
```diff
@@ -896,5 +896,6 @@ void test_set_fpenv(unsigned long env) {
   __builtin_amdgcn_set_fpenv(env);
 }
 
+// CHECK-DAG: [[$GRID_RANGE]] = !{i32 1, i32 0}
```
Comment (shiltian) [inline clang/test/CodeGenOpenCL/builtins-amdgcn.cl:899]: the upper bound is smaller than the lower bound?

Comment (arsenm) [reply clang/test/CodeGenOpenCL/builtins-amdgcn.cl:899]: Yes, this is how you are supposed to represent the wrapped set where the 0 value isn't allowed but the uint32_max is

Comment (shiltian) [reply clang/test/CodeGenOpenCL/builtins-amdgcn.cl:899]: oh I c. range does allow wrap.

**Review by shiltian — APPROVED** [review]

**Comment by ldionne** [comment]
I'm not sure what's special about your usage of Graphite, but I've seen other people use it without triggering these notifications to everyone. This is disruptive: if everyone started doing this, our Github notification system with CODEOWNERS.txt would break down completely because we'd get hundreds of spam notifications a day.

I don't use Graphite, so I don't understand what might be going wrong for certain. But this force-push seems to be the culprit:
Screenshot 2024-11-05 at 11 28 13

If you click on `Compare`, you'll see it caused the PR to affect like 3000 files, which notified everyone associated to these files. I see that you *then* updated the base branch to `main`, which reduced the diff to just *your* changes. I think you need to do it the other way around: First, change the base branch to `main`, and then rebase on top of it and force-push. That way, the diff from your branch towards the base branch will never contain more than your own changes.

**Pull request merged**